controllers/snapshot/snapshot_controller.go

```diff
@@ -96,8 +97,16 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	}
 	logger = logger.WithApp(*application)
 
-	component, err := loader.GetComponentFromSnapshot(r.Client, ctx, snapshot)
+	var component *applicationapiv1alpha1.Component
+	err = retry.OnError(retry.DefaultRetry, func(_ error) bool { return true }, func() error {
+		component, err = loader.GetComponentFromSnapshot(r.Client, ctx, snapshot)
+		return err
+	})
 	if err != nil {
+		if errors.IsNotFound(err) {
+			logger.Info(err, "Could not get component from Snapshot.  Component may have been removed.  Declining to proceed with reconciliation")
```

Comment:
`err` param cannot be used here

Reply:
Why, does `logger.Info` only allow for basic strings?

Reply:
nvm, I dug through code